### PR TITLE
Add Java and Objective-C bindings for RegisterCustomOpsUsingFunction.

### PR DESCRIPTION
**Pull request by skottmckay**
### Description

Add bindings for Android and iOS.

### Motivation and Context

Enable mobile app linking against ort-extensions library and registering the custom ops with ORT.

### CODE REVIEW

File: java/src/main/java/ai/onnxruntime/OrtSession.java [hunk 1]
```diff
@@ -701,6 +701,29 @@ public void registerCustomOpLibrary(String path) throws OrtException {
...
+     *
+     * <p>See https://onnxruntime.ai/docs/reference/operators/add-custom-op.html for more
+     * information on custom ops. See
+     * https://github.com/microsoft/onnxruntime/blob/342a5bf2b756d1a1fc6fdc582cfeac15182632fe/onnxruntime/test/testdata/custom_op_library/custom_op_library.cc#L115
+     * for an example of a custom op library registration function.
+     *
+     * @param registration_func_name The name of the registration function to call.
+     * @throws OrtException If there was an error finding or calling the registration function.
+     */
+    public void registerCustomOpsUsingFunction(String registration_func_name) throws OrtException {
```
Comment (Craigacp) [inline java/src/main/java/ai/onnxruntime/OrtSession.java:721]: Can you flip the argument from `snake_case` to `camelCase`, here and in the native function below.

File: objectivec/include/ort_session.h [hunk 2]
```diff
@@ -183,6 +183,25 @@ NS_ASSUME_NONNULL_BEGIN
                         value:(NSString*)value
                         error:(NSError**)error;
 
+/**
+ * Registers custom ops for use with {@link ORTSession}s using this SessionOptions by calling the specified
```
Comment (edgchen1) [inline objectivec/include/ort_session.h:187]: not sure if `@link` is supported by the doc renderer
https://github.com/realm/jazzy#supported-documentation-keywords

seems like `` `OrtSession` `` should work

File: objectivec/include/ort_session.h [hunk 3]
```diff
@@ -183,6 +183,25 @@ NS_ASSUME_NONNULL_BEGIN
...
+ * native function name. The custom ops library must either be linked against, or have previously been loaded
+ * by the user.
+ *
+ * The registration function must have the signature:
+ *    OrtStatus* (*fn)(OrtSessionOptions* options, const OrtApiBase* api);
+ *
+ * See https://onnxruntime.ai/docs/reference/operators/add-custom-op.html for more information on custom ops.
+ * See https://github.com/microsoft/onnxruntime/blob/342a5bf2b756d1a1fc6fdc582cfeac15182632fe/onnxruntime/test/testdata/custom_op_library/custom_op_library.cc#L115
+ * for an example of a custom op library registration function.
```
Comment (edgchen1) [inline objectivec/include/ort_session.h:196]: can we add: "Available since 1.14"?

**Review by edgchen1 — APPROVED** [review]
Objective-C API changes look good. just had some minor doc comments

File: objectivec/include/ort_session.h (hunk 3 above)
Comment (edgchen1) [reply objectivec/include/ort_session.h:203]: thanks for also updating appendExecutionProvider to be consistent!

**Review by edgchen1 — APPROVED** [review]

**Review by Craigacp — APPROVED** [review]
LGTM for the Java bits.

**Pull request merged**